code/ZAS/_gas_mixture.dm

```diff
@@ -209,8 +208,7 @@
 /datum/gas_mixture/proc/update_values()
 	total_moles = 0
 	for(var/g in gas)
-		var/moles = gas[g]
-		if(moles)
+		if(gas[g])
```

Comment:
local vars are very fast. List access is comparatively slow. I think this would be a slight performance regression.
You could turn the list access into a local var access instead.